[028,034]Test shirshov

ShirshovAlex opened the pull request:
add testContainers
add tests

ShirshovAlex committed:
add testContainers
add formTests
add internshipTests
add userTests

ShirshovAlex committed:
Delete files directory

ShirshovAlex committed:
Merge remote-tracking branch 'origin/development' into TestShirshov

# Conflicts:
#	src/main/java/by/exadel/internship/entity/Form.java

ShirshovAlex committed:
merged

ShirshovAlex committed:
change checkFirstName in FormTest

mortal-gpt commented on src/test/java/by/exadel/internship/formTest/FormTest.java:
```diff
@@ -0,0 +1,59 @@
+
+    @Test
+    public void checkListSize() throws Exception {
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.get("/forms"))
```
Where did you create testData for this test? If you reuse data from previous test, it's wrong way. Tests should be independent.

ShirshovAlex replied:
we dont create testData in this test.

mortal-gpt commented on src/test/java/by/exadel/internship/formTest/FormTest.java:
```diff
@@ -0,0 +1,59 @@
+        String content = result.getResponse().getContentAsString();
+        FormFullDTO formFullDTO = objectMapper.readValue(content, FormFullDTO.class);
+
+        assertEquals(formFullDTO.getFirstName(), "testName");
```
Are there other fields, that you can check?

ShirshovAlex replied:
yes, done

mortal-gpt commented on src/test/java/by/exadel/internship/formTest/FormTest.java:
```diff
@@ -0,0 +1,59 @@
+                MediaType.MULTIPART_FORM_DATA_VALUE, "some file".getBytes());
+        MockMultipartFile form = new MockMultipartFile("form", "form",
+                MediaType.APPLICATION_JSON_VALUE,
+                ("{\"firstName\": \"testName\",\"city\": \"string\",\"country\": \"string\"," +
```
I think it's better to move test data to separate method or class.

ShirshovAlex replied:
this testData use only once, and only in this test

mortal-gpt commented on src/test/java/by/exadel/internship/InternshipApplicationTests.java:
```diff
@@ -1,13 +1,71 @@
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.webApplicationContext).build();
+    }
+
+    protected MvcResult getResult(HttpMethod method, URI uri, ResultMatcher status) throws Exception {
```
Use this method in tests.

ShirshovAlex replied:
done

mortal-gpt commented on src/test/java/by/exadel/internship/internshipTest/InternshipTest.java:
```diff
@@ -0,0 +1,46 @@
+                .readValue(content, new TypeReference<>() {
+                });
+
+        assertEquals(guestInternshipDTO.getName(), "some name");
```
check other fields, please

ShirshovAlex replied:
add fields, done

kzaretskaya commented on src/main/resources/application.yaml:
```diff
@@ -6,8 +6,8 @@ spring:
-    password: ${SPRING_DATASOURSE_PASSWORD}
-    username: ${SPRING_DATASOURSE_USERNAME}
+    password: ${SPRING_DATASOURCE_PASSWORD}
+    username: ${SPRING_DATASOURCE_USERNAME}
```
If you change it here, change these variables in docker-compose also.

ShirshovAlex replied:
done

ShirshovAlex committed:
add more fields to check in methods

ShirshovAlex committed:
Merge branch 'development' into TestShirshov

# Conflicts:
#	pom.xml
#	src/main/java/by/exadel/internship/service/FormService.java

ShirshovAlex committed:
resolved conflicts

mortal-gpt commented on src/main/java/by/exadel/internship/dto/internshipDTO/GuestInternshipDTO.java:
```diff
@@ -40,13 +40,13 @@
 
     @NotNull
-    private Set<Skill> skills;
+    private Set<Skill> skills;//+
```
meaningless comment.

ShirshovAlex replied:
done

ShirshovAlex committed:
del comment

ShirshovAlex committed:
del comment

ShirshovAlex committed:
add @DirtiesContext to all tests

ShirshovAlex committed:
remove @DirtiesContext to all tests
remove @Testcontainers and @Containers
add static connection.start in InternshipApplicationTests
